app/bundles/PageBundle/Controller/PublicController.php

```diff
@@ -428,14 +428,16 @@ public function redirectAction($redirectId)
         $pageModel = $this->getModel('page');
         $pageModel->hitPage($redirect, $this->request);
 
-        $url = $redirect->getUrl();
+        if (null === $redirect || !$redirect->isPublished(false)) {
+            $logger->debug('Redirect with tracking_id of '.$redirectId.' not found');
+
+            $url = ($redirect) ? $redirect->getUrl() : 'unknown';
 
-        if (empty($redirect) || !$redirect->isPublished(false)) {
             throw $this->createNotFoundException($this->translator->trans('mautic.core.url.error.404', ['%url%' => $url]));
```

Comment:
Should be
```
if($url === 'unknown'){
            throw $this->createNotFoundException($this->translator->trans('mautic.core.url.error.404', ['%url%' => $url]));
}
```

Reply:
Why? That is just adding a string for the exception itself. We're throwing a 404 regardless because of this if https://github.com/mautic/mautic/pull/6579/files#diff-613422155094384a8877d07ac664791dR431